### DISCUSSION TITLE
elasticcsearch exporter dynamic trace index creation

### DISCUSSION THREAD

**Discussion by ahsanashraf918** [open-telemetry/opentelemetry-collector-contrib#27905] [Q&A]
I'd like to ingest traces into Elasticsearch and ensure that the trace index name includes the current date as a suffix. This way, a new index is created every day. I found the repository where the suffix is configured, which can be seen at the following link:

[OpenTelemetry Collector Elasticsearch Exporter - Line 99](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/elasticsearchexporter/trace_exporter.go#L99)

how can i achieve it? like test_23-Occt-2023 and so on

my values are like:

      elasticsearch:
        endpoints: [http://elasticsearch-coordinator:9200]
        trace_index: test
        traces_dynamic_index: 
          enabled: true

**Comment by leewoobin789** [comment]
it is implementable. however if you want to append date as suffix, I would recommend you to use data stream feature of elasticsearch.

**Comment by dmytro-freger-mesalvo** [comment]
Hi,

I have a question regarding log indexing in OpenTelemetry(elasticsearch). I would like to configure my logs so that a new log index is created each day, following a format like:

logs-01012025
logs-01022025
logs-01032025
etc.
I came across logs_dynamic_index and believe it might be useful for this purpose. Could you provide examples or documentation on how to configure it correctly?

Thank you for your help!
